### ISSUE TITLE
Fix S1144: Rule does not flag unused methods which match the EventHandler signature

### ISSUE THREAD

**Issue by crypto-rsa** [SonarSource/sonar-csharp#1217]
### Description

Violation of S1144 (unused private method) is not reported when the method signature matches the `EventHandler` delegate, ie. having two parameters where the second has type `EventArgs` (or is derived from it).

### Repro steps

Add the following method and make sure it is not called from anywhere
```
private void Example( object sender, EventArgs e ) { }
```

### Expected behavior

S1144 should be reported.

### Actual behavior

S1144 is not reported.

### Related information

* SonarC# Version 3.10.0.3095
* Visual Studio Version 15.6.0

**Comment by crypto-rsa** [comment]
OK, thanks for the explanation!

**Comment by valhristov** [comment]
Thanks for the understanding and I apologize for the very late investigation.
